parser-typechecker/src/Unison/Codebase/Path.hs

```diff
@@ -312,6 +312,9 @@ relativeToAncestor (Path a) (Path b) = case (a, b) of
   _ -> (empty, Path a, Path b)
 
 pattern Parent h t = Path (NameSegment h :<| t)
+pattern Cons h t <- Path (h :<| (Path -> t)) where 
```

Comment:
Instructions unclear, sorry — I suggest using the cons pattern `(<|)` that comes with Control.Lens, for which we have instances below; though if you think it’s better to have an explicit one, that’s fine by me too.

Reply:
Struggling to understand what `Parent` is there too, relative to Cons and Snoc, though I acknowledge it’s not part of this PR.

Reply:
Yeah, not sure where `Parent` is used, seems a bit weird to me. I'll try out the `<|` pattern, still want the `Empty` pattern though.

Reply:
Just wanted to point out that those instances existed, but I'm good with your choice at that point.